third_party/move/move-binary-format/src/file_format.rs

```diff
@@ -1150,6 +1150,12 @@ impl SignatureToken {
                 SignatureToken::Function(args1, results1, abs1),
                 SignatureToken::Function(args2, results2, abs2),
             ) => args1 == args2 && results1 == results2 && abs1.is_subset(*abs2),
+            (SignatureToken::Reference(ty1), SignatureToken::Reference(ty2)) => {
```

Comment:
To confirm, for vectors we need strict equality of element types?
What if you have a vector of references to function values?

Reply:
We cannot have references inside vectors. Inn general, assignability is shallow only on outer type levels and references.